src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java

```diff
@@ -55,16 +55,19 @@ public VendorResponseVO create(VendorRequestVO request) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public VendorResponseVO update(Long id, VendorRequestVO request) {
-        vendorValidator.validate(request);
-        Vendor vendor = getVendorById(id);
-        vendorMapper.update(vendor, request);
-        vendorRepository.save(vendor);
-        return vendorMapper.toVO(vendor);
+        var vendor = getVendorById(id);
+        if (!vendor.getEmail().equals(request.getEmail())) {
+            vendorValidator.validate(request);
+        }
```

Comment:
right now vendorValidator checks only email
But what if more checks will be added in the future?
then this validator will be called only in case when email was changed.

It is better to rename verndorValidator to VendorEmailValidator
by this noone will add anything else not related to email validation into this validator and this logic will be valid in the future

Reply:
fixed